app/lib/activitypub/activity/create.rb

```diff
@@ -106,7 +106,7 @@ def find_existing_status
   end
 
   def process_status_params
-    @status_parser = ActivityPub::Parser::StatusParser.new(@json, followers_collection: @account.followers_url)
+    @status_parser = ActivityPub::Parser::StatusParser.new(@json['signature'].present? ? @object : @json, followers_collection: @account.followers_url)
```

Comment:
The `signature` attribute and bearcaps are not related and this seems like a pretty arbitrary discriminator there. I think the correct fix is to always pass `@object` as… the object, but that would require changing `ActivityPub::Parser::StatusParser`'s signature to accept an extra argument.

Reply:
Fixed and commited. Please review again.

I think `@object` and `@json` should be passed at the same time, since `StatusParser` has the following code.
```ruby
as_array(@object['to'] || @json['to']).map { |x| value_or_id(x) }
```

However, there is an inconvenience: in the `StatusParser` class, there are many situations where `@object` replaces `@json`, which complicates the signature.
Indeed, `@object` used to override `@json`. I want to make sure that this is still valid now that part of `ActivityPub::Activity::Create` has been detached to `StatusParser` and is independent. No problem if it's ok for you.
https://github.com/mastodon/mastodon/commit/1060666c583670bb3b89ed5154e61038331e30c3

Reply:
Of course, I updated this code `@magic_values` to `@options`.

Reply:
Thanks! I need to give some thought about potential edge cases to be confident in the change, but it does look fine to me!